includes/class-coblocks-settings.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Register CoBlocks Settings
+ *
+ * @package   @@pkg.title
+ * @author    @@pkg.author
+ * @link      @@pkg.author_uri
+ * @license   @@pkg.license
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Main @@pkg.title Class
+ *
+ * @since 2.0.0
+ */
+class CoBlocks_Settings {
+	/**
+	 * This plugin's instance.
+	 *
+	 * @var CoBlocks_Settings
+	 */
+	private static $instance;
+
+	/**
+	 * Registers the plugin.
+	 */
+	public static function register() {
+		if ( null === self::$instance ) {
+			self::$instance = new CoBlocks_Settings();
+		}
+	}
+
+	/**
+	 * The Constructor.
+	 */
+	public function __construct() {
+
+		add_action( 'init', array( $this, 'register_settings' ) );
+
+		if ( ! get_option( 'coblocks_custom_colors_controls_enabled' ) ) {
+			add_theme_support( 'disable-custom-colors' );
+		}
+
+		if ( ! get_option( 'coblocks_gradient_presets_enabled' ) ) {
+			add_theme_support( '__experimental-editor-gradient-presets', array() );
+			add_theme_support( '__experimental-disable-custom-gradients', true );
+		}
+
+		add_action( 'init', array( $this, 'coblocks_settings_assets' ) );
+
+	}
+
+	/**
+	 * Localize CoBlock Settings Status CoBlocks settings.
+	 *
+	 * @access public
+	 */
+	public function coblocks_settings_assets() {
+		wp_localize_script(
+			'coblocks-editor',
+			'coblocksSettings',
+			array(
+				'coblocksSettings'      => get_option( 'coblocks_settings_panel_enabled' ),
```

Comment:
Localizing this option so that we can use as conditional within `../coblocks-settings/index.js`